src/hayhooks/server/utils/deploy_utils.py

```diff
@@ -389,15 +372,124 @@ def deploy_pipeline_files(
         save_files: Whether to save the pipeline files to disk
         overwrite: Whether to overwrite an existing pipeline
     """
-    pipeline_wrapper = add_pipeline_to_registry(pipeline_name, files, save_files, overwrite)
+    pipeline_wrapper = add_pipeline_wrapper_to_registry(pipeline_name, files, save_files, overwrite)
 
     if app:
-        add_pipeline_api_route(app, pipeline_name, pipeline_wrapper)
+        add_pipeline_wrapper_api_route(app, pipeline_name, pipeline_wrapper)
 
     return {"name": pipeline_name}
 
 
-def add_pipeline_to_registry(
+def deploy_pipeline_yaml(
+    pipeline_name: str,
+    source_code: str,
+    app: Optional[FastAPI] = None,
+    overwrite: bool = False,
+    options: Optional[dict[str, Any]] = None,
+) -> dict[str, str]:
+    """
+    Deploy a YAML pipeline to the FastAPI application with IO declared in the YAML.
+
+    This will add the pipeline to the registry, create flat request/response models based on
+    declared inputs/outputs, and set up the API route at /{pipeline_name}/run.
+
+    Args:
+        app: Optional FastAPI application instance. If provided, the API route will be added.
+        pipeline_name: Name of the pipeline
+        source_code: YAML pipeline source code
+        overwrite: Whether to overwrite an existing pipeline
+        options: Optional dict with additional deployment options. Supported keys:
+            - description: Optional[str]
+            - skip_mcp: Optional[bool]
```

Comment:
Seems we also support the option `save_files` looking at the code directly below